### PR TITLE
TT-38: Add get_daily_candle() convenience method to MarketDataProvider

### PR DESCRIPTION
**Pull request by xmandeng**
## Summary

Extracts a recurring manual pattern from `playground_charts.ipynb` (cell 9) into a reusable `get_daily_candle()` method on `MarketDataProvider`. The manual pattern required regex rewriting, downloading, DataFrame conversion, dict unpacking, and CandleEvent construction -- all consolidated into a single method call.

## Related Jira Issue

**Jira**: [TT-38](https://mandeng.atlassian.net/browse/TT-38)

### AC1: get_daily_candle() method added to MarketDataProvider

**File:** `src/tastytrade/providers/market.py` (line 173)

The method accepts any symbol format (`SPX`, `SPX{=m}`, `SPX{=5m}`) and rewrites to `{=d}` for daily resolution. Uses `debug_mode=True` to avoid polluting `self.frames`. Raises `ValueError` with clear message on empty results. Local import of `CandleEvent` avoids circular imports. Uses `date` type for `target_date`.

**Real Example: Symbol rewriting and candle retrieval**

```python
# Any suffix is stripped and replaced with {=d}
provider.get_daily_candle("SPX{=5m}", date(2026, 2, 10))
# -> downloads "SPX{=d}" with debug_mode=True

provider.get_daily_candle("SPX", date(2026, 2, 10))
# -> downloads "SPX{=d}" (bare symbol gets suffix appended)
```

**Results:**
- Method signature: `get_daily_candle(self, symbol: str, target_date: date) -> CandleEvent`
- Regex rewrite handles all suffix patterns: `{=d}`, `{=5m}`, `{=m}`, bare symbols
- `debug_mode=True` ensures internal `self.frames` dict is not polluted
- `ValueError` raised when no candle data available for given date
- `TYPE_CHECKING` import used for `CandleEvent` return type annotation

---

### AC2: timedelta import added

**File:** `src/tastytrade/providers/market.py` (line 5)

```python
from datetime import date, datetime, timedelta
```

Also added `TYPE_CHECKING` import from `__future__` for proper type annotation of `CandleEvent` return type without runtime circular import.

**Results:**
- `timedelta` used to construct download window: `target_date - timedelta(days=5)` to `target_date + timedelta(days=1)`
- `TYPE_CHECKING` guard used to import `CandleEvent` for return type annotation only

---

### AC3: Unit tests (3 tests, all passing)

**Real Example: Running the test suite**

```
unit_tests/providers/test_market_data_provider.py::test_get_daily_candle_returns_candle_event PASSED
unit_tests/providers/test_market_data_provider.py::test_get_daily_candle_empty_result_raises PASSED
unit_tests/providers/test_market_data_provider.py::test_get_daily_candle_bare_symbol PASSED
```

**Test details:**
- `test_get_daily_candle_returns_candle_event` -- mocks download, asserts CandleEvent returned with correct OHLCV, verifies download called with daily symbol `SPX{=d}` and `debug_mode=True`
- `test_get_daily_candle_empty_result_raises` -- mocks empty DataFrame, asserts `ValueError` raised with descriptive message
- `test_get_daily_candle_bare_symbol` -- calls with `"SPX"` (no suffix), asserts download called with `"SPX{=d}"`

**Results:**
- All 3 new tests pass
- Mocked at the `download_candle_data` boundary to isolate unit behavior
- Edge cases covered: happy path, empty results, bare symbol input

---

### AC4: Notebook updated to use new convenience method

**File:** `src/devtools/playground_charts.ipynb` (cell 9)

**Before (10-line manual pattern):**
```python
import re
daily_symbol = re.sub(r'\{=[^}]*\}', '{=d}', candle_symbol) + ('' if '{=' in candle_symbol else '{=d}')
streamer.download_candle_data(daily_symbol, start=prior_date - timedelta(days=5), ...)
df_prior = streamer.frames[daily_symbol]
row = df_prior[df_prior["time"] == prior_ts]
candle_dict = row.to_dicts()[0]
prior_day = CandleEvent(**candle_dict)
```

**After (single method call):**
```python
prior_day = streamer.get_daily_candle(candle_symbol, prior_date)
```

**Results:**
- Cell 9 reduced from 10+ lines of boilerplate to 1 line
- `import re` removed from cell 2 (no longer needed anywhere in notebook)
- Notebook logic is cleaner and more maintainable

---

## Test Evidence

- All 207 tests pass (204 existing + 3 new): `uv run pytest`
- mypy: Success, no issues found
- ruff: All checks passed
- All pre-commit hooks pass (ruff, ruff-format, mypy, trailing whitespace, end of files, debug statements)

## Changes Made

| File | Action | Purpose |
|------|--------|---------|
| `src/tastytrade/providers/market.py` | Modified | Added `get_daily_candle()` method, `timedelta` and `TYPE_CHECKING` imports |
| `unit_tests/providers/__init__.py` | New | Package init for new test directory |
| `unit_tests/providers/test_market_data_provider.py` | New | 3 unit tests for `get_daily_candle()` |
| `src/devtools/playground_charts.ipynb` | Modified | Simplified cell 9, removed unused `import re` |

[TT-38]: https://mandeng.atlassian.net/browse/TT-38?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/tastytrade/providers/market.py [hunk 1]
```diff
@@ -170,6 +173,42 @@ def coerce(dt: Union[datetime, date]) -> datetime:
...
+            symbol: Market symbol in any format (e.g. "SPX", "SPX{=m}", "SPX{=5m}").
+            target_date: The date to fetch the daily candle for.
+
+        Returns:
+            A CandleEvent with OHLCV fields populated.
+
+        Raises:
+            ValueError: If no candle data is found for the given symbol and date.
+        """
+        import re
```
Comment (xmandeng) [inline src/tastytrade/providers/market.py]: Why do we have an embedded import here?

**Review by xmandeng — COMMENTED** [review]
Please provide feedback on the embedded imports. Otherwise looks good.

File: src/tastytrade/providers/market.py (hunk 1 above)
Comment (xmandeng) [reply src/tastytrade/providers/market.py]: Good catch — moved `import re` to the top of the file in ce0f0c6 since it's stdlib with no circular risk.

The `CandleEvent` import stays local to avoid a circular dependency: `providers/market.py` is imported by modules that `messaging/models/events.py` also depends on. The `TYPE_CHECKING` guard at the top handles the type annotation, and the runtime import is deferred to method call time.
